### DISCUSSION TITLE
running on ESP32 S3 (without display)

### DISCUSSION THREAD

**Discussion by eykh** [Builty/TonexOneController#2] [General]
Hi, 
what a cool project! I had a similar idea and started fiddling with the ESP32 S3 and the ToneX One. But you are miles ahead! You mentioned "small code changes" to run your code without a display. Is it already supported? Or can you give a few hints? I want to connect the M-Vave Chocolate Controller.
Thanks!

**Comment by Builty** [comment]
Hi eykh,
Thank you.
If you don't need the display, its probably easiest to modify display.c.

You could get rid of the code inside the below functions:
void display_init(i2c_port_t I2CNum, SemaphoreHandle_t I2CMutex);
void UI_SetUSBStatus(uint8_t state);
void UI_SetBTStatus(uint8_t state);
void UI_SetPresetLabel(char* text);
void UI_SetAmpSkin(uint16_t index);
void UI_SetPresetDescription(char* text);

Leave them defined, but empty, like:
void UI_SetPresetDescription(char* text)
{
}

That way, the other parts of the code that call these functions will still compile, but it won't be trying to write to a display.
Other parts of the display.c file could be deleted.

I have been thinking about making the code enable certain parts (display, footswitches, Bluetooth) from build defines.
If I do that it will make it easier for this kind of thing.

**Reply by eykh** [reply]
Thank you! This way I will be able to help myself. Are you planning to add these compiler directives? It would be great! I could then contribute to the documentation for this headless setup (pictures, how to build).

**Reply by Builty** [reply]
I could probably do these directives. I would need some time though. I worked so hard on the app the last 2 weeks. I have coded for Tonex much more than I have played guitar with it

**Reply by eykh** [reply]
no need to hurry, we do this for fun! :) I will see what I can achieve

**Reply by Builty** [reply]
I got this done. You can now use Menu Config to enable/disable the display.

**Reply by eykh** [reply]
Thanks a lot! 
I didn't even manage to build the project yet. I will start from scratch using the instructions you linked in the README. Really cool project!!!
